src/main/java/org/sagebionetworks/web/client/widget/entity/EntityMetadata.java

```diff
@@ -59,6 +71,9 @@ public void setEntityBundle(EntityBundle bundle, Long versionNumber) {
 			view.setRestrictionPanelVisible(en instanceof TableEntity
 					|| en instanceof Folder);
 		}
+		if (en instanceof Folder || en instanceof Project) {
```

Comment:
Please move the type check to configureStorageLocation, and pass the entity in.  Then test to verify that it only asks for the upload destinations when entity is a Folder or Project.